### PR TITLE
adding error and status handling classes

### PR DESCRIPTION
**Pull request by gcmgomes**
Hello @vrod. I've implemented some initial error handling capabilities for the project, could you please take a look at it?

### CODE REVIEW

File: util/status.h [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+ * messages, this should be the go to class to use wherever executions can fail
+ * by any possible means.
+ * \code
+ * Status IsSmallerThan5(int n) {
+ *  if(n < 5) {
+ *    return Status();
+ *  }
+ *  std::stringstream error_string;
+ *  error_string << n << " is greater than or equal to 5";
+ *  return Status(error:Code::INVALID_ARGUMENT, error_string.str());
```
Comment (vrod) [inline util/status.h]: There's one missing colon here: `error:Code::INVALID_ARGUMENT`

File: util/status.h [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  UNKNOWN = 1, /** Something unforeseen happened. */
+  INVALID_ARGUMENT =
+      2, /** The provided argument is outside the method's domain. */
+};
+
+/** \brief Returns the name of the error specified by the argument.
+ *  \param code the error code that needs to be translated.
+ *  \return The translated name of the error code.
+ */
+std::string ErrorName(const Code& code);
```
Comment (vrod) [inline util/status.h]: This one should preferably return a `const std::string`

**Comment by vrod** [comment]
@gcmgomes: There is one missing colon in the example code, and one other suggested change. The rest LGTM.

File: util/status.h (hunk 1 above)
Comment (gcmgomes) [reply util/status.h]: Done. I missed colons like this around 500 times yesterday

File: util/status.h (hunk 2 above)
Comment (gcmgomes) [reply util/status.h]: Done

**Comment by gcmgomes** [comment]
Awesome, thanks!

**Pull request merged**